core/controllers/editor_test.py

```diff
@@ -1220,3 +1224,125 @@ def test_email_functionality_cannot_be_used_by_non_moderators(self):
+    DRAFT_CHANGELIST = [{
+        'cmd': 'add_state',
+        'state_name': 'State 5'}]
+    NEW_CHANGELIST = [{
+        'cmd': 'edit_exploration_property',
+        'property_name': 'title',
+        'new_value': 'New title'}]
+    INVALID_CHANGELIST = [{
+        'cmd': 'edit_exploration_property',
+        'property_name': 'title',
+        'new_value': 1}]
+
+    def setUp(self):
+        super(EditorAutosaveTest, self).setUp()
+        self.login(self.OWNER_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        # Create explorations.
+        self.save_new_valid_exploration(self.EXP_ID1, self.owner_id)
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID1)
+        exploration.add_states(['State A'])
+        exploration.states['State A'].update_interaction_id('TextInput')
+        self.save_new_valid_exploration(self.EXP_ID2, self.owner_id)
+        self.save_new_valid_exploration(self.EXP_ID3, self.owner_id)
+        # Explorations with draft set.
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.owner_id, self.EXP_ID1), user_id=self.owner_id,
+            exploration_id=self.EXP_ID1,
+            draft_change_list=self.DRAFT_CHANGELIST,
+            draft_change_list_last_updated=self.NEWER_DATETIME,
+            draft_change_list_exp_version=1).put()
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.owner_id, self.EXP_ID2), user_id=self.owner_id,
+            exploration_id=self.EXP_ID2,
+            draft_change_list=self.DRAFT_CHANGELIST,
+            draft_change_list_last_updated=self.OLDER_DATETIME,
+            draft_change_list_exp_version=1).put()
+        # Exploration with no draft.
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.owner_id, self.EXP_ID3), user_id=self.owner_id,
+            exploration_id=self.EXP_ID3).put()
+        # Go to the exploration editor page.
+        response = self.testapp.get('/create/%s' % self.EXP_ID1)
+        self.csrf_token = self.get_csrf_token_from_response(response)
+
+    def _null_method(self, *args):
+        pass
+
+    def test_draft_not_updated_newer_draft_exists(self):
+        payload = {
+            'change_list': self.NEW_CHANGELIST,
+            'version': 1,
+        }
+        apply_change_list_counter = test_utils.CallCounter(self._null_method)
+        with self.swap(exp_services, 'apply_change_list',
+                       apply_change_list_counter):
+            response = self.put_json(
+                '/createhandler/autosave_draft/%s' % self.EXP_ID1, payload,
+                self.csrf_token)
+        self.assertEqual(apply_change_list_counter.times_called, 0)
+        self.assertTrue(response['is_draft_version_valid'])
+
+    def test_draft_not_updated_validation_error(self):
+        payload = {
+            'change_list': self.INVALID_CHANGELIST,
+            'version': 1,
+        }
+        self.put_json(
```

Comment:
I've added that check here. I'm not sure which function you're referring in "Likewise below.". The only param in the response for the PUT requests is is_draft_version_valid and I'm verifying the value of that in all the test.